Update Agent Pools scope to Workspaces (#870)
hashicorp/terraform-provider-tfe: website/docs/r/agent_pool.html.markdown

* `name` - (Required) Name of the agent pool.
* `organization` - (Optional) Name of the organization. If omitted, organization must be defined in the provider config.

> * `organization_scoped` - (Optional) Whether or not the agent pool is scoped to all workspaces in the organization. Defaults to `true`.

Reviewer: ```suggestion
* `organization_scoped` - (Optional) Whether or not the agent pool can be used by all workspaces in the organization. Defaults to `true`.
```

* `name` - (Required) Name of the agent pool.
* `organization` - (Optional) Name of the organization. If omitted, organization must be defined in the provider config.
* `organization_scoped` - (Optional) Whether or not the agent pool is scoped to all workspaces in the organization. Defaults to `true`.
* `allowed_workspace_ids` - (Optional) IDs of the workspaces that to which agent pools is scoped.

>     If `organization_scoped` is set to `true`, this argument will not be served and agent pool will be scoped to all the workspaces.

Reviewer 2: `...this argument will not be served...`, this part of statement seems unclear to me, does this mean the `allowed_workspace_ids` attribute will be absent or set to null or sent to an empty slice?

Author: Removed

Author: Moved `allowed_workspace_ids` to r/agent_pool_allowed_workspaces resource.
